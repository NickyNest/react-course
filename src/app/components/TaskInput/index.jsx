import React, {Component} from 'react';
import {func} from 'prop-types';
import {Input} from 'semantic-ui-react';

class TaskInput extends Component {
    constructor(props) {
        super(props);
        this.state = {title: ''};
    }

    onChange = ({target}) => this.setState({title: target.value});

    onClick = () => {
        const {title} = this.state;
        if (!title) {
            return;
        }
        this.props.onAddTask(title);
        this.setState({title: ''});
    };

    render() {
        return (
            <div>
                <Input
                    focus
                    placeholder='add task'
                    onChange={this.onChange}
                    value={this.state.title}
                    onKeyPress={({key}) => key === 'Enter' ? this.onClick() : ''}
                    action={{primary: true, content: 'Add', onClick: this.onClick}} />
            </div>
        );
    }
}

TaskInput.propTypes = {
    onAddTask: func.isRequired
};

export default TaskInput;