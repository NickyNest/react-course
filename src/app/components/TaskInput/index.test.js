/* eslint-disable react/jsx-filename-extension */

import React from 'react';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import TaskInput from './index';

configure({adapter: new Adapter()});

describe('TaskInput', () => {
    it('TaskInput match expected snapshot', () => {
        expect(shallow(<TaskInput onAddTask={() => {}} />)).toMatchSnapshot();
    });
});