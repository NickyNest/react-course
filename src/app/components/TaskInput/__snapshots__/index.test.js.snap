// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskInput TaskInput match expected snapshot 1`] = `
<div>
  <Input
    action={
      Object {
        "content": "Add",
        "onClick": [Function],
        "primary": true,
      }
    }
    focus={true}
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="add task"
    type="text"
    value=""
  />
</div>
`;
